extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -224,6 +232,10 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 			for (let [bookModel, items] of sourcesByBook) {
 				this.bookTocManager = new BookTocManager(bookModel, targetBookModel);
 				await this.bookTocManager.updateBook(items, targetBookItem, targetSection);
+				if (this.undoTocFiles.length < this._maxUndoSize && this.undoMovedFiles.length < this._maxUndoSize) {
```

Comment:
1. We should always be keeping the latest one and instead removing the oldest entry. This looks like it'll just stop logging operations to undo once it reaches the limit unless I'm mistaken
2. I'm a little confused by the double checks. Why are you keeping separate lists here but treating them the same when checking the size? I would expect either having a single "undo operation" list that can contain different operations in it to undo (such as moving files, renaming them, etc.) or at least having each list check its own size separately